dm/syncer/error.go

```diff
@@ -61,6 +68,95 @@ func isDropColumnWithIndexError(err error) bool {
 			strings.Contains(mysqlErr.Message, "with tidb_enable_change_multi_schema is disable"))
 }
 
+// getDDLStatusFromTiDB retrieves the synchronizing status of DDL from TiDB
```

Comment:
How about move to other file but not error.go?

Reply:
Which file do you suggest?
From my perspective, the next PR to solve this issue #4689 will call GetDDLStatusFromTidb in handleSpecialDDLError, and handleSpecialDDLError is in error.go, so I put GetDDLStatusFromTidb here as well

Reply:
Maybe syncer/util.go

Reply:
Modified. Since the tests in syncer/util_test.go were using the order version of check ```pingcap/check```, I updated them using  ```stretchr/testify```.
Since there is no function in ```stretchr/testify``` has the same affect as ```HasKey```  and ```HasLen``` of Assert, I manually implement it using ```require.True``` and ```require.Equal```, respectively.

Reply:
You can use require.Len/require.Contains

Reply:
Modified.